scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,124 @@
+"""
+This script is referenced by our TerraForm config as an external data source. It
+prints the current deployment's Elasticsearch node IDs as JSON to STDOUT. If the
+list of node IDs contains fewer items than the number of placeholders in the
+Cloudwatch dashboard template, the output is padded with blanks.
+
+Note that Terraform requires the object given to an external data source is a
+map of string values, so the list of node IDs is JSON-encoded and later
+converted back into a list when processed in the Terraform config.
+"""
+import json
+import logging
+from time import (
+    sleep,
+)
+from typing import (
+    Sequence,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+import jq
+from more_itertools import (
+    padded,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to try getting the node IDs from AWS before giving up.
+max_attempts = 10
+
+
+def main() -> None:
+    node_ids = get_node_ids()
+    data = prepare_node_ids(node_ids)
+    print(json.dumps(data))
+
+
+def get_node_ids() -> Sequence[str]:
+    """
+    Return a list of ES node IDs used by the current deployment.
+    """
+    for i in range(max_attempts):
+        log.debug('Attempt %d/%d', i + 1, max_attempts)
+        try:
+            response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                   MetricName='CPUUtilization',
+                                                   RecentlyActive='PT3H',
+                                                   Dimensions=[
+                                                       {
+                                                           'Name': 'DomainName',
+                                                           'Value': config.es_domain
+                                                       }
+                                                   ])
+        except ClientError as e:
+            log.warning('%s. Retrying…', e.response['Error']['Code'])
```

Comment:
```suggestion
            log.warning('Error listing metrics (%s). Retrying …', e.response['Error']['Code'])
```

You don't control the error code (could be empty or incomprehensible) so you should always have some verbiage in front.

Please don't forget the space before the ellipsis this time.

Here and where applicable.